internal/services/attestation/attestation_provider_resource_test.go

```diff
@@ -270,3 +314,45 @@ resource "azurerm_attestation_provider" "test" {
 }
 `, template, randStr)
 }
+
+func (a AttestationProviderResource) withPolicy(data acceptance.TestData, randStr string) string {
+	// do not set `policy_signing_certificate_data`, for policy will use `jwt.SigningMethodNone`
+	template := AttestationProviderResource{}.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_attestation_provider" "test" {
+  name                = "acctestap%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  policy {
+    type = "SgxEnclave"
+    data = "%s"
+  }
+}
+`, template, randStr, a.genJWT())
+}
+
+func (AttestationProviderResource) genJWT() string {
+	// document about create policy: https://learn.microsoft.com/en-us/azure/attestation/author-sign-policy
+	policyContent := `version=1.0;
+authorizationrules
+{
+[type=="secureBootEnabled", value==true, issuer=="AttestationService"]=>permit();
+};
+
+issuancerules
+{
+=> issue(type="SecurityLevelValue", value=100);
+};`
+	b64Encoded := base64.RawURLEncoding.EncodeToString([]byte(policyContent))
+	token := jwt.NewWithClaims(jwt.SigningMethodNone, jwt.MapClaims{
+		"AttestationPolicy": b64Encoded,
+	})
+	token.Header["jku"] = "https://xxx.us.attest.azure.net/certs"
+	token.Header["kid"] = "xxx"
+	str, err := token.SignedString(jwt.UnsafeAllowNoneSignatureType)
+	_ = err
```

Comment:
we should be checking the error here?

Reply:
err should always be nil here for the input is fix.